internal-functions

```diff
@@ -75,3 +75,36 @@ fn-acl-check-app() {
     dokku_log_fail "You can only modify ACL using local dokku command on target host"
   fi
 }
+
+check-app-acl() {
+  declare desc="Checks if the current user has an ACL entry for the app"
+  declare APP=$1
+
+  if ! ( verify_app_name "$APP" 2>/dev/null ); then
+    dokku_log_fail "User $SSH_NAME does not have permissions to run $CMD on $APP, or $APP does not exist"
+  fi
+
+  ACL_FILE="$DOKKU_ROOT/$APP/acl/$SSH_NAME"
+  [[ -f "$ACL_FILE" ]] && exit 0
+
+  dokku_log_fail "User $SSH_NAME does not have permissions to run $CMD on $APP, or $APP does not exist"
+}
+
+check-service-acl() {
+  declare desc="Checks if the current user has an ACL entry for the service"
+  declare CMD=$1
+  declare SERVICE=$2
+
+  local SERVICE_TYPE="${CMD%%:*}"
+  local SERVICE_DIR="/var/lib/dokku/services/$SERVICE_TYPE/$SERVICE"
```

Comment:
@josegonzalez I'm not sure what to do here. This obviously won't work if `$PLUGIN_DATA_ROOT` has been changed to something other than `/var/lib/dokku/services/$SERVICE_TYPE`.

I'm also not sure how to test this given that `/var/lib/dokku` isn't writable in CI or in the Dokku vagrant. (See [this failure](https://travis-ci.org/dokku-community/dokku-acl/jobs/406031956) for example.)

(I'm happy to talk about this on Slack if you prefer but I thought it would be easier if you could at least see the code.)

Reply:
I think we expose that as `DOKKU_LIB_ROOT` in the `dokku` bin [here](https://github.com/dokku/dokku/blob/master/dokku#L37). Probably easy to respect that.

Reply:
Sure, but people can override that per-service, [like here for dokku-redis](https://github.com/dokku/dokku-redis/blob/daa57b7948400be03c55316626e4093a96f4d28b/config#L10). It's certainly better than hardcoding a value though.

Reply:
As an aside, these [bits](https://github.com/dokku/dokku-redis/blob/daa57b7948400be03c55316626e4093a96f4d28b/config#L4) should probably be updated to support `DOKKU_LIB_ROOT`.

I think if folks are moving these variables around, they'll have to deal with the fallout - or contribute a pr to fix :)

Reply:
Fair enough. That also provides a reasonable solution to the testing issue; I can just create a temp directory and set `DOKKU_LIB_ROOT` to that.